cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -290,28 +293,101 @@ private void copyRecursive(Path source, Path target, FileCopyMode mode) throws I
     }
   }
 
+  /**
+   * Deletes the given {@link Path} if it is a symbolic link or a Windows junction. And throws an
+   * {@link IllegalStateException} if there is a file at the given {@link Path} that is neither a symbolic link nor a
+   * Windows junction.
+   * 
+   * @param path the {@link Path} to delete.
+   * @throws IOException if the actual {@link Files#delete(Path) deletion} fails.
+   */
+  private void deleteLinkIfExists(Path path) throws IOException {
+
+    if (Files.exists(path) && Files.isSymbolicLink(path)) {
+      this.context.debug("Deleting symbolic link to be re-created at {}", path);
+      Files.delete(path);
+      return;
+    }
+    if (this.context.getSystemInfo().isWindows()) {
+      try { // since broken junctions are not detected by Files.exists(brokenJunction)
+        BasicFileAttributes attr = Files.readAttributes(path, BasicFileAttributes.class, LinkOption.NOFOLLOW_LINKS);
+        if (attr.isOther() && attr.isDirectory()) {
+          this.context.debug("Deleting symbolic link (junction) to be re-created at {}", path);
+          Files.delete(path);
+          return;
+        }
+      } catch (NoSuchFileException e) {
+        // ignore, since there is no previous file at the location, so nothing to delete
+        return;
+      }
+    }
+    throw new IllegalStateException(
+        "Failed to delete " + path + " as it is not a symbolic link or a Windows junction.");
+  }
+
+  /**
+   * Adapts the given {@link Path} to be relative or absolute depending on the given {@code relative} flag.
+   * 
+   * @param source the {@link Path} to adapt.
+   * @param targetLink the {@link Path} used to calculate the relative path to the {@code source} if {@code relative} is
+   *        set to {@code true}.
+   * @param relative the {@code relative} flag.
+   * @return the adapted {@link Path}.
+   */
+  private Path adaptPath(Path source, Path targetLink, boolean relative) {
+
+    if (relative && source.isAbsolute()) {
+      return source.toAbsolutePath();
+    }
+    if (!relative && !source.isAbsolute()) {
```

Comment:
I think the method you are looking for is `Path::toRealPath()`. During creation of link `l1`  if the source is a symlink `l2` do we want to have `l2`s location or the target of `l2` as source of the newly created link `l1`?

Reply:
I think you are locking for the method `toRealPath()`. I have implemented your mentioned behavior and wrote a test for it. Let's say link `l1` is about to be created. And as source for `l1` a path to another link `l2` is passed. Because I set `LinkOption.NOFOLLOW_LINKS` the source of `l1` will be `l2` instead of the source of `l2`.